### PR TITLE
#22 Implementer les endpoints de l'objet StatusRequete

### PR DESCRIPTION
**Pull request by MentalistHB**
I have removed some code lines which are not needed for now like and have ensured that the integration tests are still working.

### CODE REVIEW

**Comment by Coccoonx** [comment]
I did my review. Changes are good. Nothing to update. Waiting @Chombon review to merge into master.

File: backend/src/main/java/com/ambacam/model/StatusRequete.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+@Entity
+public class StatusRequete implements Serializable {
+
+	private static final long serialVersionUID = -8019398584857375323L;
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@NotEmpty(message = "The name must be empty")
```
Comment (Coccoonx) [inline backend/src/main/java/com/ambacam/model/StatusRequete.java]: message =  "The name must **not** be empty"

File: backend/src/main/java/com/ambacam/rest/roles/RolesResource.java [hunk 2]
```diff
@@ -29,17 +29,14 @@
...
 	@Autowired
 	private RoleService roleService;
-	@Autowired
-	RoleResource roleResource;
 
 	@RequestMapping(method = RequestMethod.POST)
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces(MediaType.APPLICATION_JSON)
-	public Role create(@RequestBody @Valid Role entity) {
+	public Role create(@RequestBody @Valid Role create) {
```
Comment (Coccoonx) [inline backend/src/main/java/com/ambacam/rest/roles/RolesResource.java]: Concerning the name of the incoming object, I used to keep the name of the class as variable name. Eg
`public Role create(@RequestBody @Valid Role role) {}`
What do you think ?

File: backend/src/main/java/com/ambacam/rest/statusrequetes/StatusRequetesResource.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	private static final Logger log = LoggerFactory.getLogger(StatusRequetesResource.class);
+
+	@Autowired
+	private StatusRequeteService statusRequeteService;
+
+	@RequestMapping(method = RequestMethod.POST)
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	public StatusRequete create(@RequestBody @Valid StatusRequete create) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/rest/statusrequetes/StatusRequetesResource.java]: If in you change create->role in RolesResource please change here also create->statusRequete

File: backend/src/test/java/com/ambacam/rest/statusrequetes/StatusRequetesResourceIT.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+	}
+
+	@Test
+	public void updateNotFound() {
+		StatusRequete update = buildStatusRequete();
+		given().contentType(ContentType.JSON).body(update).put(ApiConstants.STATUS_REQUETE_ITEM, random.nextLong())
+				.then().log().body().statusCode(404);
+	}
+
```
Comment (Chombon) [inline backend/src/test/java/com/ambacam/rest/statusrequetes/StatusRequetesResourceIT.java]: Please add the test UpdateSameName()

Comment (Chombon) [reply backend/src/test/java/com/ambacam/rest/statusrequetes/StatusRequetesResourceIT.java]: Can you please add also for Role?

**Comment by MentalistHB** [comment]
Nice code review, guys!!! I apply the comments right now and push again.

File: backend/src/test/java/com/ambacam/rest/statusrequetes/StatusRequetesResourceIT.java (hunk 4 above)
Comment (MentalistHB) [reply backend/src/test/java/com/ambacam/rest/statusrequetes/StatusRequetesResourceIT.java]: Sure!

File: backend/src/main/java/com/ambacam/rest/roles/RolesResource.java (hunk 2 above)
Comment (MentalistHB) [reply backend/src/main/java/com/ambacam/rest/roles/RolesResource.java]: That is right, I agree with you @Coccoonx . I think, it is a good practice.

**Comment by MentalistHB** [comment]
The comments have already been applied as well as the squash and rebase onto the master.

**Comment by Chombon** [comment]
The code is ok for me. @Coccoonx I am waiting for your approval to merge

**Pull request merged**